test/Microsoft.Net.Http.Headers.Tests/StringWithQualityHeaderValueTest.cs

```diff
@@ -124,6 +124,9 @@ public void Parse_SetOfValidValueStrings_ParsedCorrectly()
             CheckValidParse("t ;  q  =   0", new StringWithQualityHeaderValue("t", 0));
             CheckValidParse("iso-8859-5", new StringWithQualityHeaderValue("iso-8859-5"));
             CheckValidParse("unicode-1-1; q=0.8", new StringWithQualityHeaderValue("unicode-1-1", 0.8));
+            CheckValidParse("quality_limit; q=000000001", new StringWithQualityHeaderValue("quality_limit", 1));
+            CheckValidParse("quality_limit; q=0000001.0", new StringWithQualityHeaderValue("quality_limit", 1));
+            CheckValidParse("quality_limit; q=1.0000000", new StringWithQualityHeaderValue("quality_limit", 1));
```

Comment:
In fact `TryParseQualityDouble()` accepts it but it is rejected [later](https://github.com/aspnet/HttpAbstractions/blob/dev/src/Microsoft.Net.Http.Headers/BaseHeaderParser.cs#L60).